nova_vm/src/engine/bytecode/executable.rs

```diff
@@ -2044,278 +2046,68 @@ impl CompileEvaluation for ast::ObjectPattern<'_> {
         );
 
         for ele in &self.properties {
-            match &ele.key {
-                ast::PropertyKey::StaticIdentifier(identifier) => {
-                    let identifier_string = ctx.create_identifier(&identifier.name);
-                    if ele.shorthand {
-                        if let ast::BindingPatternKind::AssignmentPattern(_) = &ele.value.kind {
-                            todo!();
+            assert!(!ele.computed, "TODO: Complex object patterns");
+
+            if ele.shorthand {
+                let ast::PropertyKey::StaticIdentifier(identifier) = &ele.key else {
+                    unreachable!()
+                };
+                assert!(matches!(
+                    &ele.value.kind,
+                    ast::BindingPatternKind::BindingIdentifier(_)
+                ));
+                let identifier_string = ctx.create_identifier(&identifier.name);
+                ctx.exe.add_instruction_with_identifier(
+                    Instruction::BindingPatternBind,
+                    identifier_string,
+                );
+            } else {
+                let key_string = match &ele.key {
+                    ast::PropertyKey::StaticIdentifier(identifier) => {
+                        String::from_str(ctx.agent, &identifier.name).into_value()
+                    }
+                    ast::PropertyKey::NumericLiteral(literal) => {
+                        let numeric_value = Number::from_f64(ctx.agent, literal.value);
+                        if let Number::Integer(_) = numeric_value {
+                            numeric_value.into()
```

Comment:
**nitpick:** `into_value()` to be explicit about the conversion target and fit with the rest of the code.